tap-version-14-specification.md

```diff
@@ -0,0 +1,741 @@
+
+For example:
+
+```
+1..3
+ok 1
+not ok 2
+ok 3
+```
+
+If there is no number the harness must maintain its own counter until
+the script supplies test numbers again. So the following test output
+
+```
+1..6
+not ok
+ok
+not ok
+ok
+ok
+```
+
+has five tests. The sixth is missing.  For example, in Perl,
+`Test::Harness` will generate
+
+```
+FAILED tests 1, 3, 6
+Failed 3/6 tests, 50.00% okay
+```
+
+- Description
+
+An optional description may be provided, separated from the ok/not-ok
+and test number by one or more space characters, terminated by either
+a `#` character, or the end of the line.
+
+If the Description starts with a hyphen `-` and any amount of
+whitespace, it SHOULD be discarded by the Harness.  For example, these
+test lines should all be identical:
+
+```
+ok 5 some description
+ok 5 - some description
+ok 5 - some description # and some comment
+```
+
+The harness may do whatever it wants with the description.
+
+- Directive
+
+The test point may include a directive, following a hash on the test
+line.
+
+There are currently two directives allowed: TODO and SKIP. These are
+discussed below.  Any other directive is ignored.
+
+#### Test Line Summary
+
+To summarize:
+
+- ok/not ok (required)
+- Test number (recommended)
+- Description (recommended)
+- Directive (only when necessary)
+
+The regular expression pattern for a test line is:
+
+```
+/^(not )?ok( +[1-9][0-9]*)?( +(- +)[\u0001-\u0009\u000B-\u000C\u000E-\u0022\u0024-\u00FF]+)?( #(.*))?$/
+```
+
+### YAML blocks
+
+If a test line is immediately followed by a 2-space indented block
+beginning with the line `/^ {2}---$/` and ending with the line
+`/^ {2}\.\.\.$/` that block will be interpreted as an inline YAML
+document.
+
+The YAML encodes a data structure that provides more detailed
+information about the preceding failed test.  The YAML document is
+indented with 2 spaces to make it visually distinct from the
+surrounding test results and to make it easier for the parser to
+recover if the trailing `/^ {2}\.\.\.$/` terminator is missing.
+
+For example:
+
+```
+not ok 3 Resolve address
+  ---
+  message: "Failed with error 'hostname peebles.example.com not found'"
+  severity: fail
+  data:
+    got:
+      hostname: 'peebles.example.com'
+      address: ~
+    expected:
+      hostname: 'peebles.example.com'
+      address: '85.193.201.85'
+  ...
+```
+
+The corresponding data structure in JSON would look like this:
+
+```json
+{
+  "message": "Failed with error 'hostname peebles.example.com not found'",
+  "severity": "fail",
+  "data": {
+    "got": {
+      "hostname": "peebles.example.com",
+      "address": null
+    },
+    "expected": {
+      "hostname": "peebles.example.com",
+      "address": "85.193.201.85"
+    }
+  }
+}
+```
+
+If a line occurs within the YAML document that is not indented, or if
+the TAP stream ends before the `/^ {2}\.\.\.$/` terminator line, or if the
+YAML document does not parse as YAML, then it MUST be treated as
```

Comment:
Munging YAML to make it valid seems pretty bad, surely the "handled by the Harness the same as any other non-TAP lines" leaves if vague anyway?